### DISCUSSION TITLE
Specifying a Lambert Conformal Grid in MET

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#2301] [Configuration] [answered]
Hi:

I'm trying to regrid data to lambert conformal in grid_stat using a grid specification string.  [Appendix B](https://met.readthedocs.io/en/latest/Users_Guide/appendixB.html#appendix-b-map-projections-grids-and-polylines) gives the information about specifying a lambert grid as lambert Nx Ny lat_ll lon_ll lon_orient D_km R_km standard_lat_1 [ standard_lat_2 ] N|S.  However, I've been unable to get the expected grid output and I'm not sure where I am going wrong.  I've verified that MET can read this grid (by regridding outside of MET and then using plot_data_plane).

Here is the grid information:
Projection: PROJ_LAMBERT_CONF
  origin_lon (deg): 120
  origin_lat (deg): 10
  lat1 (deg)      : 10
  lat2 (deg)      : 40
  nx, ny, nz: 300, 325, 1
  minx, miny, minz: -130, 1200, 0
  maxx, maxy: 468, 1848
  dx, dy, dz: 2, 2, 1
  grid origin latitude: 10
  grid origin longitude: 120
  minLat, minLon: 20.9214, 118.642
  maxLat, maxLon: 27.0258, 124.887

Expected output in MET:
Projection = "Lambert Conformal" ;
		:hemisphere = "N" ;
		:scale_lat_1 = "10.000000" ;
		:scale_lat_2 = "40.000000" ;
		:lat_pin = "10.000000" ;
		:lon_pin = "120.000000" ;
		:x_pin = "65.000000" ;
		:y_pin = "-600.000000" ;
		:lon_orient = "120.000000" ;
		:d_km = "2.000000" ;
		:r_km = "6371.200000" ;
		:nx = "300" ;
		:ny = "325 grid_points" 

I've tried different specification strings, but the most recent is:
GRID_STAT_REGRID_TO_GRID = lambert 300 325 20.9214 118.642 120.0 2.0 6371.2 10.0 40.0 N

And it produces this grid in MET:
Projection = "Lambert Conformal" ;
		:hemisphere = "N" ;
		:scale_lat_1 = "10.000000" ;
		:scale_lat_2 = "40.000000" ;
		:lat_pin = "20.921400" ;
		:lon_pin = "118.642000" ;
		:x_pin = "0.000000" ;
		:y_pin = "0.000000" ;
		:lon_orient = "120.000000" ;
		:d_km = "2.000000" ;
		:r_km = "6371.200000" ;
		:nx = "300" ;
		:ny = "325 grid_points" ;

The lat_pin, lon_pin, x_pin, and y_pin are not what they should be.  Am I specifying something incorrectly or can this grid just not be specified using a string in MET?  Any information would be helpful.

Thanks,
Tina

**Comment by JohnHalleyGotway** [comment, marked as answer]
@CPKalb I took a look at the source code and understand what's going on. Not every grid definition used by MET is reproducible via a grid specification string.

You're trying to reproduce a grid definition that uses non-zero values for the x_pin and y_pin values:
```
:x_pin = "65.000000" ;
:y_pin = "-600.000000" ;
```
Lambert grids defined by a grid specification string are parsed by the this function:
```
bool parse_lambert_grid(const StringArray &grid_strings, Grid & g)
```
And on this [line of code](https://github.com/dtcenter/MET/blob/3f0411b65cef5719bf11b6fb75f7e8413fe633f0/src/libcode/vx_grid/find_grid_by_name.cc#L337), you'll see that x_pin and y_pin are set to 0:
```
ldata.x_pin = 0.0;
ldata.y_pin = 0.0;
```
That is to say that there's no way to set a grid specification string and produce any non-zero pin location.

The question is what do these x_pin/y_pin/lat_pin/lon_pin settings mean? Depending on how the lambert grid is specified, we have a reference latitude and longitude location. That might be the lower-left corner, the midpoint, or perhaps some other location. The x_pin and y_pin values define the offset from the lower-left corner to the location of the reference lat/lon values.

Since the [lambert string](https://met.readthedocs.io/en/latest/Users_Guide/appendixB.html#grid-specification-strings) defines the lat/lon of the lower-left corner, you'll always get pin values of 0.

Of course, two grids can be *equivalent*, meaning that are comprised of the same set of lat/lon locations. But MET only considers them to be *equal* if the data structures defining them are identical.

I'll note that when we switch over to defining grids using string from the **Proj** library, this should get simpler.

**Reply by CPKalb** [reply]
@JohnHalleyGotway, thanks.  I suspect my grids are not equivalent.  My original data shows reflectivity values in the domain.  But, when I use the lambert string, I get no data in the domain.  Specifically, it's all 0s.  Is there any way to know how to produce an equivalent grid with a lambert string?

Tina

**Reply by JohnHalleyGotway** [reply]
Can you point to some specifics... like a specific NetCDF file that's on the target grid?

If it contains a field of latitude and longitude values, you could just look at the lat/lon of the lower-left corner and use that in the grid specification string with the (0,0) x/y pin location.

**Reply by CPKalb** [reply]
I ended up regridding a file using another method.  There, I was able to identify the lower left lat/lon points and use those in the grid specification string to get the expected results.  In this case, the lower left lat/lon are not the minimum latitude and longitude, which made it tricky to figure out what the lower left points should be without an actual gridded file to work from.
